pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -1150,8 +1151,15 @@ func traceNextInfoReq(ctx *zedagentContext) bool {
 	if !isNettraceEnabled(ctx) {
 		return false
 	}
+	uptime := time.Since(ctx.startTime)
+	lastNetdumpAge := time.Since(ctx.lastInfoNetdumpPub)
+	// Ensure we get at least one netdump for the currently tested EVE upgrade.
+	if zboot.IsCurrentPartitionStateInProgress() &&
+		(ctx.lastInfoNetdumpPub.IsZero() || lastNetdumpAge > uptime) {
```

Comment:
What will happen if EVE will reboot during download in process?

Reply:
Netdump is collected after download completes. So if interrupted, EVE would restart download and netdump would be collected afterwards. Note that for downloader microservice we always collect netdumps - this change is for zedagent and nim to collect at least one netdump for requests towards the controller during upgrade. With reboot during reboot we may end up collecting more, but that is OK. Not having any netdump for failed upgrade would be a problem.

Reply:
@uncleDecart Will you please re-approve? Somehow adding comments makes eden tests skipped :/

Reply:
Great and in case of reboot it'll be 2 different logs which we will be able to differentiate